Sources/SnapshotTesting/AssertSnapshot.swift

```diff
@@ -328,3 +329,21 @@ func sanitizePathComponent(_ string: String) -> String {
     .replacingOccurrences(of: "\\W+", with: "-", options: .regularExpression)
     .replacingOccurrences(of: "^-|-$", with: "", options: .regularExpression)
 }
+
+private class TestObserver: NSObject, XCTestObservation {
+    private static var registered = false
+    private static var registerQueue = DispatchQueue(label: "co.pointfree.SnapshotTesting.testObserver")
+
+    static func registerIfNeeded() {
+        registerQueue.sync {
+            if !registered {
+                registered = true
+                XCTestObservationCenter.shared.addTestObserver(TestObserver())
+            }
+        }
+    }
+
+    func testCaseDidFinish(_ testCase: XCTestCase) {
+        counterMap = [:]
```

Comment:
I think this should use `counterQueue`

Reply:
Yes, I should! Thanks for spotting this.